korok/rerankers/rerankers.py

```diff
@@ -1,18 +1,30 @@
+import importlib
 from typing import List, Sequence, Tuple
-
-from sentence_transformers import CrossEncoder
-
+    
 
 class CrossEncoderReranker:
+    """A reranker class that uses a cross-encoder to rerank the results."""
+
     def __init__(self, model_name: str = "cross-encoder/ms-marco-MiniLM-L-6-v2", max_length: int = 512) -> None:
         """
         Initialize the CrossEncoder reranker.
 
         :param model_name: The name of the pre-trained cross-encoder model.
         :param max_length: The maximum sequence length for the cross-encoder.
         """
+        if not self.is_available():
+            raise ImportError("sentence_transformers is not installed. Please install it using `pip install korok[rerankers]`.")
+        else:
+            global CrossEncoder
```

Comment:
I'm not sure if the global is necessary here. Do we need it outside of the scope of this function?

Reply:
Well, you can also pass it around inside the class via `self.CrossEncoder`, but usually imports inside functions are limited to the function scope.